src/utils.c

```diff
@@ -0,0 +1,29 @@
+#include "string.h"
+#include "stdbool.h"
+
+bool is_valid_fqdn_name(const char *name)
+{
+    int count = 0;
+    int allowed;
+    for (unsigned long i = strlen(name); i > 0; i--)
+    {
+        if (name[i] == '.')
+        {
+            count = 0;
+            continue;
+        }
+        allowed = (name[i] >= '0' && name[i] <= '9') ||
```

Comment:
According to RFC1035, if doesn't consider IDN, a domain allow letter, digit and dash. You check function seems not allow dash.